core/Tracker/TrackerCodeGenerator.php

```diff
@@ -65,24 +73,38 @@ public function generate($idSite, $piwikUrl, $mergeSubdomains = false, $groupPag
                     continue;
                 }
 
-                $options .= '  _paq.push(["setCustomVariable", ' . $index++ . ', "' . $visitorCustomVariable[0] . '", "' . $visitorCustomVariable[1] . '", "visit"]);' . PHP_EOL;
+                $options .= sprintf(
+                    '  _paq.push(["setCustomVariable", %d, %s, %s, "visit"]);%s',
+                    $index++,
+                    json_encode($visitorCustomVariable[0]),
+                    json_encode($visitorCustomVariable[1]),
```

Comment:
Is it possible for `$visitorCustomVariable` to have non-string elements? If so, perhaps they should be casted to strings.